Sources/muterCore/CLICommands/RunCommand/Steps/DiscoverSourceFiles.swift

```diff
@@ -48,40 +47,54 @@ private extension DiscoverSourceFiles {
         ]
     }
     
-    func discoverSourceFiles(inDirectoryAt path: FilePath,
-                             excludingPathsIn providedExcludeList: [FilePath] = []) -> [FilePath] {
+    private func discoverSourceFiles(
+        inDirectoryAt rootPath: FilePath,
+        excludingPathsIn providedExcludeList: [FilePath] = []
+    ) -> [FilePath] {
         let excludeList = providedExcludeList + defaultExcludeList
-        let subpaths = fileManager.subpaths(atPath: path) ?? []
+        let subpaths = fileManager.subpaths(atPath: rootPath) ?? []
+        
         return includeSwiftFiles(
             from: subpaths
-                .exclude(pathsContainingItems(from: excludeList))
-                .map(curry(append)(path))
+                .exclude(pathsContainingItems(root: rootPath, from: excludeList))
+                .map { append(root: rootPath, to: $0) }
         )
     }
     
-    func pathsContainingItems(from excludeList: [FilePath]) -> (FilePath) -> Bool {
-        return { (path: FilePath) in
-            
-            for item in excludeList where path.contains(item) {
+    private func pathsContainingItems(root: FilePath, from excludeList: [FilePath]) -> (FilePath) -> Bool {
+        let excludeAlso: [FilePath] = excludeList.reduce(into: []) { result, path in
```

Comment:
I think this could be rewritten as a `flatMap`¹ and be simpler.

```swift
let excludeAlso: [FilePath] = excludeList.flatMap { path in
    path.contains("*")
        ? expandGlobExpressions(root: root, pattern, pattern)
        : path
}
```

¹ I can never remember whether it's `flatMap` or `compactMap` any more. You get the gist.